geps/gep-1897.md

```diff
@@ -125,7 +141,326 @@ This GEP is the outcome of the TLS use cases #4 and #5 in
+substituted without blocking acceptance of the content of the API change.
+
+The selection of the applicable Gateway API persona is important in the design of TLSConnectionPolicy, because it provides
+a way to explicitly describe the _expectations_ of the connection to the application.  TLSConnectionPolicy is configured
+by the application developer Gateway API persona to signal what the application developer _expects_ in connections to
+the application, from a TLS perspective.  Only the application developer can know what the application expects, so it is
+important that this configuration be managed by that persona.
+
+During the course of discussion of this proposal, we did consider allowing the cluster operator persona to have some access
+to Gateway cert validation, but as mentioned, TLSConnectionPolicy is used primarily to signal what the application
+developer expects in the connection.  Granting this expectation to any other role would blur the lines between role
+responsibilities, which violates the role-oriented design principle of Gateway API. As mentioned in Non-goal #7,
+providing a mechanism for the cluster operator gateway role to override gateway to backend TLS settings is not covered
+by this proposal, but should be addressed in a future update.  One idea is to use two types: ApplicationTLSConnectionPolicy,
+and GatewayTLSConnectionPolicy, where the application developer is responsible for the former, the cluster operator is
+responsible for the latter, and the cluster operator may configure whether certain settings may be overridden by
+application developers.
+
+Using the viewpoint of the application running inside the Pods, the TLSConnectionPolicy should contain configuration
+for these use cases of the Gateway as a TLS Client:
+
+- An explicit signal that TLS should be used by this connection.
+- Which Certificate Authority (CA) to use, whether the system bundled CA, an internal CA, or self-signed.
+- TLS version to expect.  This allows implementations to introduce features to disallow or force certain TLS versions.
+
+TLSConnectionPolicy is defined as a Direct Policy Attachment without defaults or overrides, applied to a Service that
+accesses the backend in question, where the TLSConnectionPolicy resides in the same namespace as the Service it is
+applied to.  The TLSConnectionPolicy and the Service must reside in the same namespace in order to prevent the
+complications involved with sharing trust across namespace boundaries.  By choosing the Service resource rather than the
+Route resource, we can reuse the same TLSConnectionPolicy for all the different Routes that might point to this Service.
+For the use case where certificates are stored in their own namespace, users may create Secrets and use ReferenceGrants
+for a TLSConnectionPolicy-to-Secret binding.  Implementations must respect a ReferenceGrant for cross-namespace Secret
+sharing to TLSConnectionPolicy, even if they don't for other cross-namespace sharing.
```

Comment:
I'm still not sure I understand what the Secrets are doing in this - is there any real system where a server gives the client a certificate to use ? Normal authentication is that each workload ( including gateway ) has an identity that it presents - there are some mechanism in the handshake to allow some choices, but I have not seen this in many mTLS implementations.

Reply:
We are not addressing mTLS in this proposal.

This section discusses cross-namespace sharing, and the requirement for TLSConnectionPolicy and Service to be in the same namespace.  For the use case where certificates are stored in their own, separate namespace, users may create Secrets and use ReferenceGrants.